streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -85,8 +85,23 @@
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
+    default void register(final StateStore store,
```

Comment:
I understand the basic logic here, but that also means in the deprecated init of the state stores (those with the ProcessorContext) the stores will not be able to register the callback. Why don't we keep the checkpoint callback in both ProcessorContext::register and the StateStoreContext::register and mark the one in ProcessorContext deprecated?

Reply:
That's correct. The deprecated init method will never be invoked if the new one is implemented, so there's really no reason to implement both. In our classes that have both, I think it's usually because the new init method is delegating to the old one, but this would be a good opportunity to just drop the code in the old one.

Reply:
I see what you're getting at, but the whole point of the deprecation is that users should be migrating to the new API asap. We're going to drop that deprecated init pretty soon anyway, so I'd rather be a bit more hard-line about not adding new things to it in the mean time.